eth_abi/codec_packed.py

```diff
@@ -0,0 +1,41 @@
+from typing import (
+    Any,
+    Iterable,
+    Tuple,
+)
+import warnings
+
+from eth_typing.abi import (
+    Decodable,
+    TypeStr,
+)
+
+from eth_abi.codec import (
+    ABICodec,
+)
+from eth_abi.registry import (
+    ABIRegistry,
+    registry_packed as default_packed_registry,
+)
+
+warnings.warn(
+    "Packed mode encoding is an experimental feature.  Please report any "
+    "problems at https://github.com/ethereum/eth-abi/issues."
+)
+
+
+class ABICodecPacked(ABICodec):
+    def __init__(self, registry: ABIRegistry=None):
+        if registry is None:
+            ABICodec.__init__(self, default_packed_registry)
+        else:
+            ABICodec.__init__(self, registry)
+
+    def is_encodable(self, typ: TypeStr, arg: Any) -> bool:
+        raise NotImplementedError("'is_encodable' is not  supported in packed mode")
```

Comment:
I haven't looked too deeply at the packed stuff. Why can't we test if a value is encodable?

Reply:
Was just keeping with the methods that were previously implemented for packed encoding. I can dig into this some more and see. 🤔

Reply:
I think the lack of an `is_encodable_packed` function may have just been an oversight or for the sake of expedience.  I just looked through the code and all of the `Packed*` encoder classes seem to inherit from base classes which implement `validate_value`.  So they should all have what is needed to run an encodability check.  Otherwise, I can't think of another reason that it wouldn't be possible.  Now that we have the `ABICodec` class, we should get this for "free."  We just need delete the `NotImplementedError` override and add something like `is_encodable_packed = default_codec_packed.is_encodable`.

Reply:
Why rename the method with the `_packed` suffix? We don't rename any of the other methods like `decode_abi_packed`.

I'd say just implement it once in the base codec class and get it for free with the same name.

Reply:
I think he's suggesting that because we're trying to stay passive by exporting the methods in `eth_abi/packed`. Like:
```
encode_abi_packed = default_codec_packed.encode_abi
encode_single_packed = default_codec_packed.encode_single
```
Since those were previously exported.

Reply:
Added the method here:
https://github.com/ethereum/eth-abi/pull/109/commits/abceb187b537e3b754356feb3bbfba6a7989684a#diff-f7c24c7b7a7b6405352611f3ee624887

Need to add some tests now for it.